src/script/api/script_depotlist.cpp

```diff
@@ -16,6 +16,7 @@
 
 ScriptDepotList::ScriptDepotList(ScriptTile::TransportType transport_type)
 {
+	EnforceDeityOrCompanyModeValid();
```

Comment:
I know, I've fixed it locally.

I'm working on splitting this PR in smaller, hopefully easier to review PRs as it has become a bit of a mess in this branch. Like first a PR for the trivial refactor, and then update this PR to add all the missing ones with logic to prevent them going missing in the future.